Fix broken links in Updating and Upgrading guide

adamlazik1 opened the pull request:
Fixed two broken links in Updating and Upgrading guide in section 3.1.2,
procedure steps 9 and 10. Split step 10 into sections for EL 7 and EL 8
and added corresponding link in the EL 8 section.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [x] Foreman 3.1/Katello 4.3

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_disconnected_satellite.adoc:
```diff
@@ -82,9 +82,16 @@ Reboot these hosts after the upgrade has completed.
 ----
 
-. Obtain the latest ISO files by following the https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html-single/installing_satellite_server_from_a_disconnected_network/installing-satellite-server-disconnected#downloading-the-binary-dvd-images_satellite[Downloading the Binary DVD Images] procedure in the _Installing {ProjectServer} from a Disconnected Network_ g ...
+. Obtain the latest ISO files by following the https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html-single/installing_satellite_server_in_a_disconnected_network_environment/index#downloading-the-binary-dvd-images_satellite[Downloading the Binary DVD Images] procedure in the _Installing {ProjectServer} in a Disconnected Network Environment_ guide.
```
While you're at it, I think this (and some others below) can be changed to:
```suggestion
. Obtain the latest ISO files by following the {BaseURL}installing_satellite_server_in_a_disconnected_network_environment/index#downloading-the-binary-dvd-images_satellite[Downloading the Binary DVD Images] procedure in the _Installing {ProjectServer} in a Disconnected Network Environment_ guide.
```

adamlazik1 commented:
@ekohl Thank you for your suggestion. Incorporated the suggested change. Also updated additional references to 'Installing {ProjectServer} in a Disconnected Network Environment' guide.

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_disconnected_satellite.adoc:
```diff
@@ -22,7 +22,7 @@ endif::[]
 
 * Review and update your firewall configuration before upgrading your {ProjectServer}.
-For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html-single/installing_satellite_server_from_a_disconnected_network/index#satellite-ports-and-firewalls-requirements_satellite[Ports and Firewalls Requirements] in _Installing {ProjectServer} from a Disconnected Network_.
+For more information, see {BaseURL}installing_satellite_server_from_a_disconnected_network/index#satellite-ports-and-firewalls-requirements_satellite[Ports and Firewalls Requirements] in _Installing {ProjectServer} in a Disconnected Network Environment_ guide.
```
Apologies for the partial review earlier, but we even have `InstallingDisconnectedDocURL` which is even more precise.
```suggestion
For more information, see {InstallingDisconnectedDocURL}satellite-ports-and-firewalls-requirements_satellite[Ports and Firewalls Requirements] in _Installing {ProjectServer} in a Disconnected Network Environment_ guide.
```

In https://github.com/theforeman/foreman-documentation/pull/856 I intended to clean it up, but never got the traction to get merged. I still think it's a good idea so I rebased that. I don't mind rebasing later if this is merged, but also feel free to pull in the first commit of that.

adamlazik1 replied:
No worries, I will get right on it.

adamlazik1 commented:
Updated links with the suggested attribute. Also found and fixed two additional links.

ekohl reviewed (approved):
:+1: from a technical point of view. @maximiliankolb mind doing a check?

ekohl commented:
> * [x]  Foreman 3.1/Katello 4.3

Note that Satellite on 3.1 doesn't support EL8. It needs a separate cherry pick PR.

adamlazik1 commented:
@ekohl
> Note that Satellite on 3.1 doesn't support EL8. It needs a separate cherry pick PR.
I would like to confirm this. On https://github.com/theforeman/foreman-documentation/milestone/2 is written:
> - Foreman 3.1: both upstream and Satellite can run on EL7 & EL8

maximiliankolb commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_disconnected_satellite.adoc:
```diff
@@ -22,7 +22,7 @@ endif::[]
 
 * Review and update your firewall configuration before upgrading your {ProjectServer}.
-For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html-single/installing_satellite_server_from_a_disconnected_network/index#satellite-ports-and-firewalls-requirements_satellite[Ports and Firewalls Requirements] in _Installing {ProjectServer} from a Disconnected Network_.
+For more information, see {InstallingDisconnectedDocURL}Ports_and_Firewalls_Requirements_satellite[Ports and Firewalls Requirements] in _Installing {ProjectServer} in a Disconnected Network Environment_.
```
This hardcoded context is really weird but IMHO correct. I assume this is due to disconnected being officially unsupported for Foreman/Katello.

```
➜  foreman-documentation git:(master) git grep -n "Ports_and_Firewalls_Requirements_{context}"
guides/common/modules/ref_ports-and-firewall-requirements.adoc:1:[id="Ports_and_Firewalls_Requirements_{context}"]

➜  foreman-documentation git:(master) cat guides/doc-Installing_Server_Disconnected/master.adoc | grep "^:context:"
:context: satellite
```

ekohl replied:
> I assume this is due to disconnected being officially unsupported for Foreman/Katello.

Note that I think it could be supported, but the instructions aren't there. And repositories are completely different between the various build flavors.

adamlazik1 replied:
Anything for me to do here?

maximiliankolb replied:
No, this is ok!

maximiliankolb commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_disconnected_satellite.adoc:
```diff
@@ -281,7 +288,7 @@ Review the results and address any highlighted error conditions before performin
 +
 If the script fails due to missing or outdated packages, you must download and install these separately.
-For more information, see the https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html-single/installing_satellite_server_from_a_disconnected_network/installing-satellite-server-disconnected#resolving-package-dependency-errors_satellite[Resolving Package Dependency Errors] section in the _Installing {ProjectServer} from a Disconnected Network_ guide.
+For more information, see the {InstallingDisconnectedDocURL}resolving-package-dependency-errors_satellite[Resolving Package Dependency Errors] section in the _Installing {ProjectServer} in a Disconnected Network Environment_ guide.
```
```suggestion
For more information, see {InstallingDisconnectedDocURL}resolving-package-dependency-errors_satellite[Resolving Package Dependency Errors] in _Installing {ProjectServer} in a Disconnected Network Environment_.
```

Maybe it'd be a good idea to add `InstallingDisconnectedDocTitle` as attribute to `guides/common/attributes-titles.adoc`.

adamlazik1 replied:
Hello! I added the attribute in a separate commit. Do you suggest squashing it to into the first commit or should I leave it as it is?

maximiliankolb replied:
Oh, I think I made an error here: `InstallingServerDisconnectedDocTitle` is already defined in `guides/common/attributes-titles.adoc`. So suggesting `InstallingDisconnectedDocTitle` was wrong on my behalf. Sorry for causing confusion!

It would probably be best to someday move this to "attributes-base.adoc", but that's not something that needs to be done within this PR.

adamlazik1 replied:
Understood. I updated the commits with the correct attribute.

maximiliankolb reviewed (commented):
Made some commects which are not blocking your PR. If you want to go the extra mile, add a title for the guide.

maximiliankolb commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_disconnected_satellite.adoc:
```diff
@@ -82,9 +82,16 @@ Reboot these hosts after the upgrade has completed.
+. Create directories to serve as a mount point, mount the ISO images, and configure the `rhel7-server` or the `rhel8` repository:
++
+.For {EL} 8
+Follow the {InstallingDisconnectedDocURL}configuring-the-base-operating-system-with-offline-repositories-in-rhel-8_satellite[Configuring the Base Operating System with Offline Repositories in RHEL 8] procedure in _{InstallingServerDisconnectedDocTitle}_.
```
If you want, you could use `{RHEL}` instead of RHEL (same below). your call.

adamlazik1 replied:
I have no problem making the change, but shouldn't we reference the sections by their exact names?

maximiliankolb replied:
yeah, that makes sense. So no change necessary.

maximiliankolb reviewed (commented):
change looks good to me.

I wish we'd rename "InstallingDisconnectedDocURL" to "InstallingServerDisconnectedDocURL" at some point because it does not match the DocTitle attribute. This is unrelated to your PR.

maximiliankolb commented:
Merged to master and cherry-picked both commits: ✅ 
   05bcbbcd9..9497c8b53  3.4 -> 3.4
   a34b849e3..36d80f747  3.3 -> 3.3
   9f1a6b086..a3cae3bb8  3.2 -> 3.2
   d2f9c2cdc..17fc77e88  3.1 -> 3.1
